scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -45,17 +45,24 @@
   [node-name->resource-map resource-keyword]
   (->> node-name->resource-map vals (map resource-keyword) (reduce +)))
 
+(defn total-gpu-resource
+  "Given a map from node-name->resource-keyword->amount,
+  returns a map from gpu model to count for all nodes."
+  [node-name->resource-map]
+  (->> node-name->resource-map vals (map :gpus) (apply merge-with +)))
+
 (defn generate-offers
   "Given a compute cluster and maps with node capacity and existing pods, return a map from pool to offers."
   [compute-cluster node-name->node node-name->pods]
   (let [compute-cluster-name (cc/compute-cluster-name compute-cluster)
         node-name->capacity (api/get-capacity node-name->node)
         node-name->consumed (api/get-consumption node-name->pods)
-        node-name->available (pc/map-from-keys (fn [node-name]
-                                                 (merge-with -
-                                                             (node-name->capacity node-name)
-                                                             (node-name->consumed node-name)))
-                                               (keys node-name->capacity))]
+        node-name->available (api/deep-merge-with - node-name->capacity node-name->consumed)
+        ; Grab every unique GPU model being represented so that we can set counters for capacity and consumed for each GPU model
+        gpu-models (->> node-name->capacity vals (map :gpus) (apply merge) keys set)
+        total-gpu-capacity (total-gpu-resource node-name->capacity)
```

Comment:
We are only using this code for the counters. The counters are compute-cluster specific by design. Since the metrics are compute-cluster specific, these lines should be living in the compute cluster code (api code is compute cluster agnostic)

Reply:
renamed to `gpu-metric-capacity` and `gpu-metric-consumed` for more clarification